module/VuFind/src/VuFind/View/Helper/Root/MakeTag.php

```diff
@@ -0,0 +1,277 @@
+        'base',
+        'br',
+        'col',
+        'embed',
+        'hr',
+        'img',
+        'input',
+        'link',
+        'meta',
+        'param',
+        'source',
+        'track',
+        'wbr'
+    ];
+
+    /**
+     * Render an HTML tag
+     *
+     * A string passed into $attrs will be treated like a class.
+     * These two are equivalent:
+     * > MakeTag('div', 'Success!', 'alert alert-success')
+     * > MakeTag('div', 'Success!', ['class => 'alert alert-success'])
+     *
+     * Additional options
+     * - escapeContent: Default true, set to false to skip escaping (like for HTML).
+     *
+     * @param string       $tagName  Element tag name
+     * @param string       $contents Element contents (must be properly-formed HTML)
+     * @param string|array $attrs    Tag attributes (associative array or class name)
+     * @param array        $options  Additional options
+     *
+     * @return string HTML for an anchor tag
+     */
+    public function __invoke(
+        string $tagName,
+        string $contents,
+        $attrs = [],
+        $options = []
+    ) {
+        // $attrs not an object, interpret as class name
+        if (!is_array($attrs)) {
+            $attrs = !empty($attrs) ? ['class' => $attrs] : [];
+        }
+
+        // Compile attributes
+        return $this->compileTag($tagName, $contents, $attrs, $options);
+    }
+
+    /**
+     * Verify HTML tag matches HTML spec
+     *
+     * @param string $tagName Element tag name
+     *
+     * @return void
+     * @throws InvalidArgumentException
+     */
+    protected function verifyTagName(string $tagName)
+    {
+        // Simplify check by making tag lowercase
+        $lowerTagName = mb_strtolower($tagName, 'UTF-8');
+
+        // Existing tag?
+        if (in_array($lowerTagName, $this->validBodyTags)) {
+            return;
+        }
+
+        // Check if it's a valid custom element
+        // Spec: https://html.spec.whatwg.org/#autonomous-custom-element
+
+        // All valid characters for a Potential Custom Element Name
+        // Concated for clarity (space not a valid character)
+        $PCENChar = '[' .
+            '\-\.0-9_a-z' .
+            '\x{B7}' .
+            '\x{C0}-\x{D6}' .
+            '\x{D8}-\x{F6}' .
+            '\x{F8}-\x{37D}' .
+            '\x{37F}-\x{1FFF}' .
+            '\x{200C}-\x{200D}' .
+            '\x{203F}-\x{2040}' .
+            '\x{2070}-\x{218F}' .
+            '\x{2C00}-\x{2FEF}' .
+            '\x{3001}-\x{D7FF}' .
+            '\x{F900}-\x{FDCF}' .
+            '\x{FDF0}-\x{FFFD}' .
+            '\x{10000}-\x{EFFFF}' .
+            ']*';
+
+        // First character must be a letter (uppercase or lowercase)
+        // Needs one hyphen to designate custom element, more groups valid
+        $validCustomTagPattern = '/^[a-z]' . $PCENChar . '(\-' . $PCENChar . ')+$/u';
+
+        // Is valid custom tag?
+        if (!preg_match($validCustomTagPattern, strtolower($lowerTagName))) {
```

Comment:
Why are we `strtolower`-ing a string that's already lowercase?

Reply:
removed and tested and passed